# Is Tesla Overpriced?

In our previous notebook, we created a dataframe with our key financial ratios that we will use to analyse Tesla's shareprice. 

To visualise our data, we will use plotly's interactive charts.

The ratios used are based on investopedias article on determining over or undervalued stocks found here:

https://www.investopedia.com/articles/investing/101316/how-tell-if-stock-overvalued-or-undervalued.asp

The Ratios we are interested in are:

- P/E, P/B, and EV/EBITDA Ratios 
- Price-to-Sales ratio (P/S)
- Price-to-Dividend (P/D)
- Enterprise Value-to-Sales (EV/S) Ratio 
- Price/Earnings to Growth (PEG) Ratio

## Reading And Exploring The Dataframe

References:

https://www.investopedia.com/terms/p/price-to-salesratio.asp

https://www.investopedia.com/terms/p/price-to-bookratio.asp

https://www.investopedia.com/terms/e/ev-ebitda.asp

In [16]:
import numpy as np
import pandas as pd

import chart_studio.plotly as py
import plotly.graph_objs as go
from plotly.subplots import make_subplots
from plotly.offline import iplot, init_notebook_mode
import plotly.express as px

import cufflinks
cufflinks.go_offline(connected=True)
init_notebook_mode(connected=True)


In [17]:
df = pd.read_csv("company_fin_ratios.csv", index_col=0)
df.head()

ticker              company                  sector  price_to_book_ratio  \
0   TSLA                Tesla  Consumer Discretionary            17.138617   
1    MMM           3M Company             Industrials             9.318123   
2    ABT  Abbott Laboratories             Health Care             5.070602   
3   ABBV          AbbVie Inc.             Health Care           -14.820874   
4   ABMD          ABIOMED Inc             Health Care            16.684694   

   price_to_sales_ratio  price_earnings_ratio  price_earnings_to_growth_ratio  \
0              4.614833           -131.581636                       23.860429   
1              2.917858             20.518221                        2.213313   
2              4.940913             42.754239                       31.921239   
3              3.640840             15.366174                        4.001636   
4             20.315925             60.350413                       24.163300   

   dividend_yield  enterprise_value_multiple  
0             NaN                  48.094311  
1        0.035364                  14.359891  
2        0.014400                  22.755484  
3        0.052561                   9.830543  
4       -0.000000                  64.179470

In [18]:
df.describe()

price_to_book_ratio  price_to_sales_ratio  price_earnings_ratio  \
count           496.000000            496.000000            496.000000   
mean             -0.124999              5.859739             18.967322   
std             132.915949             52.840341            304.078772   
min           -1991.763043           -624.518114          -6465.525181   
25%               1.662408              1.305301             13.130291   
50%               3.115174              2.769245             20.627869   
75%               6.052460              5.088207             32.427480   
max            1353.873275            877.076883           1098.688175   

       price_earnings_to_growth_ratio  dividend_yield  \
count                      496.000000      418.000000   
mean                        22.688210      123.515450   
std                        155.407551     2521.151982   
min                       -994.385555       -0.030412   
25%                          2.016578        0.012299   
50%                          5.003917        0.021353   
75%                         11.036849        0.031399   
max                       2100.063424    51545.261580   

       enterprise_value_multiple  
count                 499.000000  
mean                   38.625719  
std                   505.920358  
min                  -387.705882  
25%                     9.607426  
50%                    13.812940  
75%                    19.333914  
max                 11287.556316

In [19]:
df.info()

<class 'pandas.core.frame.DataFrame'>
Int64Index: 503 entries, 0 to 502
Data columns (total 9 columns):
ticker                            503 non-null object
company                           503 non-null object
sector                            503 non-null object
price_to_book_ratio               496 non-null float64
price_to_sales_ratio              496 non-null float64
price_earnings_ratio              496 non-null float64
price_earnings_to_growth_ratio    496 non-null float64
dividend_yield                    418 non-null float64
enterprise_value_multiple         499 non-null float64
dtypes: float64(6), object(3)
memory usage: 39.3+ KB


In [20]:
df.shape

(503, 9)

In [21]:
df['sector'].value_counts()

Information Technology    71
Industrials               69
Financials                66
Consumer Discretionary    64
Health Care               60
Consumer Staples          33
Real Estate               31
Materials                 28
Utilities                 28
Energy                    27
Communication Services    26
Name: sector, dtype: int64

## Visualising Our Data With Plotly

We have over 500 companies in our dataset which will make visual comparisons challenging. We will therefore look to filter and visualise by sector.


## Subplotting Our Data

In [35]:
cd = df[df['sector'] == 'Consumer Discretionary']

fig = make_subplots(
    rows=6, cols=2,
    subplot_titles=('P/B Ratio', 'Frequency Distributions (%)', 
                    'P/S Ratio', '', 
                    'P/E Ratio', '',
                    'Price Earnings To Growth Ratio', '',
                   'Dividend Yield', '',
                   'Enterprise Value Multiple', ''))

fig.add_trace(go.Scatter(
    x=cd['ticker'],
    y=cd['price_to_book_ratio'],
    mode='markers',
    name='P/B'),
    row=1,col=1)

fig.add_trace(go.Histogram(
    x=cd['price_to_book_ratio'],
    histnorm='percent',
    name='P/B'),
    row=1, col=2)

fig.add_trace(go.Scatter(
    x=cd['ticker'],
    y=cd['price_to_sales_ratio'],
    mode='markers',
    name='P/S'),
    row=2,col=1)

fig.add_trace(go.Histogram(
    x=cd['price_to_sales_ratio'],
    histnorm='percent',
    name='P/S'),
    row=2, col=2)

fig.add_trace(go.Scatter(
    x=cd['ticker'],
    y=cd['price_earnings_ratio'],
    mode='markers',
    name='P/E'),
    row=3,col=1)

fig.add_trace(go.Histogram(
    x=cd['price_earnings_ratio'],
    histnorm='percent',
    name='P/E'),
    row=3, col=2)

fig.add_trace(go.Scatter(
    x=cd['ticker'],
    y=cd['price_earnings_to_growth_ratio'],
    mode='markers',
    name='PE/G'),
    row=4,col=1)

fig.add_trace(go.Histogram(
    x=cd['price_earnings_to_growth_ratio'],
    histnorm='percent',
    name='PE/G'),
    row=4, col=2)

fig.add_trace(go.Scatter(
    x=cd['ticker'],
    y=cd['dividend_yield'],
    mode='markers',
    name='Yield'),
    row=5,col=1)

fig.add_trace(go.Histogram(
    x=cd['dividend_yield'],
    histnorm='percent',
    name='Yield'),
    row=5, col=2)

fig.add_trace(go.Scatter(
    x=cd['ticker'],
    y=cd['enterprise_value_multiple'],
    mode='markers',
    name='EMM'),
    row=6,col=1)

fig.add_trace(go.Histogram(
    x=cd['enterprise_value_multiple'],
    histnorm='percent',
    name='EMM'),
    row=6, col=2)


fig.update_xaxes(showticklabels=False, col=1)
fig.update_layout(showlegend=False,width=1000,
    height=1500)

In [38]:
cd.describe()


price_to_book_ratio  price_to_sales_ratio  price_earnings_ratio  \
count            63.000000             63.000000             63.000000   
mean              5.894515              1.992853             31.203343   
std              18.950001              1.535871             97.499705   
min             -79.954564              0.225224           -131.581636   
25%               1.954954              0.834305             12.147589   
50%               3.900057              1.598084             19.627489   
75%               7.654794              2.800421             26.349066   
max              81.352146              7.775772            747.074043   

       price_earnings_to_growth_ratio  dividend_yield  \
count                       63.000000       47.000000   
mean                        10.121459        0.025022   
std                        174.144991        0.015684   
min                       -994.385555        0.000489   
25%                          2.327145        0.013969   
50%                          4.417267        0.021856   
75%                          7.372028        0.037158   
max                        807.010633        0.068038   

       enterprise_value_multiple  
count                  63.000000  
mean                    7.568683  
std                    51.176402  
min                  -387.705882  
25%                     8.991932  
50%                    12.577219  
75%                    16.024887  
max                    48.094311

## Price To Sales Ratio

In [37]:
fig = px.scatter(df[df['sector'] == 'Consumer Discretionary'], 
                 x='ticker', 
                 y='price_to_book_ratio',
                 hover_name = 'company',
                 labels={'price_to_book_ratio': 'P/B Ratio',
                        'ticker': 'Company'},
                 

                 )

fig.update_xaxes(showticklabels=False)
fig.update_layout(title_text='Consumer Discretionary Sector - P/B Ratio')

fig.show()

Investopedia: The price-to-book ratio compares a company's market value to its book value. The market value of a company is its share price multiplied by the number of outstanding shares. The book value is the net assets of a company.

The P/B ratio has been favored by value investors for decades and is widely used by market analysts. Traditionally, any value under 1.0 is considered a good P/B for value investors, indicating a potentially undervalued stock. However, value investors may often consider stocks with a P/B value under 3.0 as their benchmark.

Out of the 64 comapanies in this sector. Tesla has the 5th highest P/B ratio at 17.14 which is considerably higher than the traditional rule of thumb. and is in the top quartile of the frequency distribution although not an outlier.

Tesla's P/E ratio shows that their share price approx. 17 times higher than it's book value which would suggest that the comapanies fundamentals do not support the high share price.



## Price To Sales Ratio

Investopedia: The price-to-sales (P/S) ratio is a valuation ratio that compares a company’s stock price to its revenues. It is an indicator of the value placed on each dollar of a company’s sales or revenues.

The ratio shows how much investors are willing to pay per dollar of sales. Like all ratios, the P/S ratio is most relevant when used to compare companies in the same sector. A low ratio may indicate the stock is undervalued, while a ratio that is significantly above the average may suggest overvaluation.

The P/S ratio doesn’t take into account whether the company makes any earnings or whether it will ever make earnings.

In [25]:
fig = px.scatter(df[df['sector'] == 'Consumer Discretionary'], 
                 x='ticker', 
                 y='price_to_sales_ratio',
                 hover_name = 'company',
                 labels={'price_to_sales_ratio': 'P/S Ratio',
                        'ticker': 'Company'},

                 )

fig.update_xaxes(showticklabels=False)

fig.show()

We once again see a the same trend when comparing Tesla's P/S ratio with other companies in this sector. Given that Tesla is a growth company and currently lacks consistent profits the P/S ratio would be a more realistic metric when analysing Tesla's current share price. The relatively high P/S ratio would however suggest that Tesla's stock price per dollar of sales would not offer good value for someone with a value investing strategy.

## Enterprise Value Multiple

In [40]:
fig = px.scatter(df[df['sector'] == 'Consumer Discretionary'], 
                 x='ticker', 
                 y='enterprise_value_multiple',
                 hover_name = 'company',
                 labels={'enterprise_value_multiple': 'EV Multiple',
                        'ticker': 'Company'},

                 )

fig.update_xaxes(showticklabels=False)

fig.show()

We will disregard Newell Brands as it is a clear outlier which skews our data.

Investopedia: Enterprise multiple, also known as the EV multiple, is a ratio used to determine the value of a company. The enterprise multiple looks at a firm in the way that a potential acquirer would by considering the company's debt. Stocks with an enterprise multiple of less than 7.5x based on the last 12 months (LTM) is generally considered a good value. However, using a strict cutoff is generally not appropriate because this is not an exact science.

Unlike many other common measures, the enterprise multiple takes into account a company's debt and cash levels in addition to its stock price and relates that value to the firm's cash profitability (e.g. the price-to-earnings [P/E] ratio).

Investors mainly use a company's enterprise multiple to determine whether a company is undervalued or overvalued. A low ratio indicates that a company might be undervalued and a high ratio indicates that the company might be overvalued.

Tesla's EV multiple of 48 makes it the highest EV multiple of any company in the consumer discretionary sector. Higher EV multiples are generally seen in higher growth industries which is indicative of Tesla and their stage in the business life cycle.


## Findings

We started this analysis with Elon Musk's quote suggesting that Tesla's was stock price was overvalued. After comparing fundamental ratios with companies in the consumer discretionary sector of the S&P 500. We haven't found any evidence of underlying value that would make Tesla an attractive choice for value investors.

Tesla's rapid share price growth is indicative of a speculative bubble and given the growing competition in the electric car market by established brands, it is questionable whether Tesla will be able to ramp up production and become profiatable given the resources of their competitors.

Of course Tesla's brand currently holds the same status as the most popular tech brands but how long will this continue.

## Next Steps

- Comparison with other auto brands i.e VW etc.
- Share Price comparison with other autobrands using ochl data.